### PR TITLE
add callbackWaitsForEmptyEventLoop on handler wrappers

### PR DESCRIPTION
**Pull request by pocesar**
* remove unnecessary await from returns

I think, since the wrappers are acting as bridges to the real handlers, it's safe to assume `context.callbackWaitsForEmptyEventLoop = false`, since the code will most likely be using `async (event, context) => Promise<any>` instead of `(event, context, callback) => void`

This has bitten me in the past (on unrelated projects), with lambda timing out, even though it had completed. 
Some issues in the wild:
https://www.serverlesslife.com/AWS_Serverless_Common_Mistakes_Communication_with_Other_Systems.html#connections
https://github.com/awslabs/aws-serverless-express/issues/134
https://medium.com/@hichaelmart/reviewing-the-aws-lambda-nodejs10-x-runtime-84de1e500aac
https://www.jeremydaly.com/reuse-database-connections-aws-lambda/
https://stackoverflow.com/a/41623101/647380

I understand if the desired behavior should default to letting the consumer decide, feel free to disregard, but since the code does async things before eventually calling the underlaying handler, it might catch early cases of errors. gotta say, it's a weird behavior, for sure

### CODE REVIEW

**Comment by sam-goodwin** [comment]
If Promises are used to handle the request, then the event loop should always be empty when we return the result, right? Or do DB connections have issues? 

Perhaps we should make this configurable on a per-function basis?

**Comment by pocesar** [comment]
the issue I see is that when using variables that "hang around" outside of the handler, it *might* give varying results. on my test code, I'm trying to connect to mongodb atlas from the handler, and I also have external db connection 'cached variables' (as per documentation). this might be true for things that run longer than the handler itself, but databases are a prime example

**Comment by sam-goodwin** [comment]
You should implement `Dependency` for your Mongo DB database and use the `depends:` logic to bootstrap your runtime handler with access to the DB.

Related: We should also support AWS's managed MongoDB database. This would be easy once the CDK implements their L2 constructs for it.

File: packages/punchcard/lib/lambda/function.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ export class Function<T, U, D extends Dependency<any>> implements Entrypoint, Re
       const requestMapper: Mapper<T, any> = this.request === undefined ? Raw.passthrough() : Raw.forShape(this.request);
       const responseMapper: Mapper<U, any> = this.response === undefined ? Raw.passthrough() : Raw.forShape(this.response);
       return (async (event: any, context: any) => {
+        context.callbackWaitsForEmptyEventLoop = false;
```
Comment (sam-goodwin) [inline packages/punchcard/lib/lambda/function.ts]: This could be set in the `bootstrap` function of a MongoDB database dependency.

Comment (pocesar) [reply packages/punchcard/lib/lambda/function.ts]: indeed! good idea, it's never good to lie/omit about your dependencies 👍

Comment (sam-goodwin) [reply packages/punchcard/lib/lambda/function.ts]: It also means that the flag is only set for functions who use the dependency - better encapsulation.

**Pull request closed without merging**